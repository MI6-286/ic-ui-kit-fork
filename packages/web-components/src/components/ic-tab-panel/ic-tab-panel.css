@import "../../global/normalize.css";

:host(.ic-tab-panel-hidden) {
  display: none;
}

:host(:not(.ic-tab-panel-hidden)) {
  display: block;
}

:host {
  color: var(--ic-color-text-primary);
}
