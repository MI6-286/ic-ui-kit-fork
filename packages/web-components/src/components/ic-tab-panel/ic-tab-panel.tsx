import {
  Component,
  Element,
  Event,
  EventEmitter,
  Host,
  Prop,
  h,
} from "@stencil/core";
import { IcThemeMode } from "../../utils/types";

@Component({
  tag: "ic-tab-panel",
  styleUrl: "ic-tab-panel.css",
  shadow: true,
})
export class TabPanel {
  @Element() el: HTMLIcTabPanelElement;

  /**
   * @internal If `true`, the tab panel will be displayed.
   */
  @Prop() active: boolean = false;

  /**
   * @internal Determines whether black variant of the tabs should be displayed.
   */
  @Prop() monochrome?: boolean = false;

  /**
   * @internal The shared ID that links the panel and tab.
   */
  @Prop({ reflect: true }) panelId?: string;

  /**
   * @internal The shared ID of the currently selected tab.
   */
  @Prop() selectedTab?: string;

  /**
   * @internal The position of the tab panel inside the tabs array in context.
   */
  @Prop({ reflect: true }) tabPosition?: number;

  /** @internal Determines whether the light or dark variant of the tabs should be displayed. */
  @Prop() theme?: IcThemeMode = "inherit";

  /**
   * @internal Emitted when a tab panel is dynamically created.
   */
  @Event() tabPanelCreated: EventEmitter<HTMLIcTabPanelElement>;

  /**
   * @internal Emitted when a tab panel is unmounted.
   */
  @Event() tabPanelRemoved: EventEmitter<void>;

  connectedCallback(): void {
    this.tabPanelCreated.emit(this.el);
  }

  render() {
    const { active, theme } = this;
    return (
      <Host
        class={{
          [`ic-theme-${theme}`]: theme !== "inherit",
          "ic-tab-panel-hidden": !active,
        }}
        role="tabpanel"
        aria-hidden={`${!active}`}
      >
        <div>
          <slot></slot>
        </div>
      </Host>
    );
  }
}
