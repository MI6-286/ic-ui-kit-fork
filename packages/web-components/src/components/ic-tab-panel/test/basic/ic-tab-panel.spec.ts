import { newSpecPage } from "@stencil/core/testing";
import { TabPanel } from "../../ic-tab-panel";

describe("ic-tab-panel component", () => {
  it("should render correctly when it is the active tab panel", async () => {
    const page = await newSpecPage({
      components: [TabPanel],
      html: `<ic-tab-panel active>IC Tab Test</ic-tab-panel>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render correctly when not the active tab panel", async () => {
    const page = await newSpecPage({
      components: [TabPanel],
      html: `<ic-tab-panel>IC Tab Test</ic-tab-panel>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
