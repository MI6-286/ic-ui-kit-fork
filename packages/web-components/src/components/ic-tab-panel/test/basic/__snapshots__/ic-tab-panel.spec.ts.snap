// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-tab-panel component should render correctly when it is the active tab panel 1`] = `
<ic-tab-panel active="" aria-hidden="false" role="tabpanel">
  <mock:shadow-root>
    <div>
      <slot></slot>
    </div>
  </mock:shadow-root>
  IC Tab Test
</ic-tab-panel>
`;

exports[`ic-tab-panel component should render correctly when not the active tab panel 1`] = `
<ic-tab-panel aria-hidden="true" class="ic-tab-panel-hidden" role="tabpanel">
  <mock:shadow-root>
    <div>
      <slot></slot>
    </div>
  </mock:shadow-root>
  IC Tab Test
</ic-tab-panel>
`;
